pkg/chaosctl/cmd/recover.go

```diff
@@ -0,0 +1,178 @@
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "register completion func for flag `namespace`")
+	}
+	err = recoverCmd.RegisterFlagCompletionFunc("label", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "register completion func for flag `label`")
+	}
+	return recoverCmd, nil
+}
+
+func recoverResourceCommand(option *RecoverOptions, chaosType string, builder recover.RecoverBuilder) *cobra.Command {
+	return &cobra.Command{
+		Use:   fmt.Sprintf(`%s (PODs) [-n NAMESPACE]`, chaosType),
+		Short: fmt.Sprintf(`Recover %s from certain pods`, chaosType),
+		Long:  fmt.Sprintf(`Recover %s from certain pods`, chaosType),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, cancel, err := common.CreateClient(context.TODO(), managerNamespace, managerSvc)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+			return option.Run(builder(client), client, args)
+		},
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			if len(args) != 0 {
+				return []string{}, cobra.ShellCompDirectiveNoFileComp
+			}
+			client, cancel, err := common.CreateClient(context.TODO(), managerNamespace, managerSvc)
+			if err != nil {
+				common.PrettyPrint(errors.Wrap(err, "create client").Error(), 0, common.Red)
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+			defer cancel()
+			return option.List(client)
+		},
+	}
+}
+
+// Run recover
+func (o *RecoverOptions) Run(recover recover.Recover, client *ctrlclient.CtrlClient, args []string) error {
+	pods, err := o.selectPods(client, args)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, pod := range pods {
+		err = recover.Recover(ctx, pod)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// List pods to recover
+func (o *RecoverOptions) List(client *ctrlclient.CtrlClient) ([]string, cobra.ShellCompDirective) {
+	pods, err := o.selectPods(client, []string{})
+	if err != nil {
+		common.PrettyPrint(errors.Wrap(err, "select pods").Error(), 0, common.Red)
+	}
+
+	var names []string
+	for _, pod := range pods {
+		names = append(names, pod.Name)
+	}
+
+	return names, cobra.ShellCompDirectiveNoFileComp
+}
+
+func (o *RecoverOptions) selectPods(client *ctrlclient.CtrlClient, names []string) ([]*ctrlclient.PartialPod, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	selector := v1alpha1.PodSelectorSpec{
+		GenericSelectorSpec: v1alpha1.GenericSelectorSpec{
+			Namespaces: []string{o.namespace},
+		},
+	}
+
+	labelSelector := map[string]string{}
+	if o.labels != nil && len(*o.labels) > 0 {
+		for _, label := range *o.labels {
+			pair := strings.Split(label, "=")
+			if len(pair) == 2 {
```

Comment:
if not, the label is invalid,  should we print a log?

Reply:
Sure